providers/websockets/coinbase/ws_data_handler.go

```diff
@@ -0,0 +1,132 @@
+package coinbase
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"go.uber.org/zap"
+
+	"github.com/skip-mev/slinky/oracle/config"
+	"github.com/skip-mev/slinky/providers/base/websocket/handlers"
+	providertypes "github.com/skip-mev/slinky/providers/types"
+	oracletypes "github.com/skip-mev/slinky/x/oracle/types"
+)
+
+const (
+	// Name is the name of the exchange.
+	Name = "coinbase_websocket"
+)
+
+var _ handlers.WebSocketDataHandler[oracletypes.CurrencyPair, *big.Int] = (*WebSocketDataHandler)(nil)
+
+// WebSocketDataHandler implements the WebSocketDataHandler interface. This is used to
+// handle messages received from the Coinbase websocket API.
+type WebSocketDataHandler struct {
+	logger *zap.Logger
+
+	// config is the config for the Coinbase web socket API.
+	cfg config.ProviderConfig
+
+	// Sequence is the current sequence number for the Coinbase web socket API per currency pair.
+	sequence map[oracletypes.CurrencyPair]int64
+}
```

Comment:
this should be threadsafe since there is only one of these handlers running per web socket provider so no mutex is necessary